plugins/outputs/groundwork/groundwork.go

```diff
@@ -244,15 +266,19 @@ func (g *Groundwork) parseMetric(metric telegraf.Metric) (string, *transit.Dynam
 				Label:      value.Key + "_cr",
 				Value: &transit.TypedValue{
 					ValueType:   transit.DoubleType,
-					DoubleValue: critical,
+					DoubleValue: &critical,
 				},
 			})
 		}
 
-		typedValue := new(transit.TypedValue)
-		err := typedValue.FromInterface(value.Value)
-		if err != nil {
-			return "", nil, err
+		typedValue := transit.NewTypedValue(value.Value)
+		if typedValue == nil {
+			g.Log.Warnf("could not convert value, skipping field %s: %v", value.Key, value.Value)
```

Comment:
I think there was a version that also logged what the value type was?

I also think the support is lacking for `uint` types.